scripts/azureml-assets/azureml/assets/config.py

```diff
@@ -535,23 +534,27 @@ def __init__(self, storage_name: str, container_name: str, container_path: str):
             None,
         )
 
-    @property
-    def uri(self) -> str:
-        """Asset URI."""
-        # If we have already cached the URI, then simply return it
-        if self._uri is not None:
-            return self._uri
+    def get_uri(self, token_expiration_timedelta=datetime.timedelta(hours=1)) -> str:
+        """Get Asset URI.
+
+        :param token_expiration_timedelta: Amount of time until token expiration
+        :type token_expiration_timedelta: datetime.timedelta
+        """
+        # There are 3 values token can be set to
+        # None = no attempts have been made to create a SAS token
+        # Empty string = container is public (no SAS token required)
+        # Neither of the above = the SAS token
 
         # Build the simple, non-SAS URI for quick use later in the function.
         # Its possible that the account URI may need additional tweaking to add a SAS
         # token if the account does not allow for anonymous access.
-        self._uri = f"{self._account_uri}/{self._container_name}/{self._container_path}"
+        uri = f"{self._account_uri}/{self._container_name}/{self._container_path}"
 
         # If a SAS token has been explicitly set, then assume that the URI+token is valid.
         # Simply append the token and return. If no SAS token is set, then proceed to rest of function.
         if self.token is not None:
-            self._uri += "?" + self.token
-            return self._uri
+            uri += "?" + self.token
+            return uri

```

Comment:
This should handle the empty string:
```suggestion
            if self.token:
                uri += "?" + self.token
            return uri
```